io/PtsReader.cpp

```diff
@@ -63,27 +63,34 @@ void PtsReader::initialize(PointTableRef table)
 
     // Very first line is point count
     Utils::fromString(buf, m_PointCount);
-    Utils::closeFile(m_istream);
-}
-
-
-void PtsReader::addDimensions(PointLayoutPtr layout)
-{
-    // Dimensions are fixed in PTS
 
+    // Peek second line to determine dimensions to add.
+    // Allows reading of PTS files with only X Y Z dimensions.
+    // Only allow 3 fields or 7 fields.
     m_dims.push_back(Dimension::Id::X);
     m_dims.push_back(Dimension::Id::Y);
     m_dims.push_back(Dimension::Id::Z);
-    m_dims.push_back(Dimension::Id::Intensity);
-    m_dims.push_back(Dimension::Id::Red);
-    m_dims.push_back(Dimension::Id::Green);
-    m_dims.push_back(Dimension::Id::Blue);
-
-    for (auto d: m_dims)
+    std::getline(*m_istream, buf);
+    StringList fields = Utils::split2(buf, m_separator);
```

Comment:
My concern here is that the second line doesn't always contain the fields.  If you take a look at the (supposed) PTS file in #1508, the field line looks like a comment in the first line.

I guess knowing more about why this came up (your file source), would be valuable.